core/src/edu/cornell/gdiac/main/NPCModel.java

```diff
@@ -0,0 +1,327 @@
+
+    float width = json.getFloat("width") * (1 / drawScale.x);
+    float height = json.getFloat("height") * (1 / drawScale.y);
+    setDimension(width, height);
+
+    JsonValue properties = json.get("properties").child();
+    Color debugColor = null;
+    int debugOpacity = -1;
+    while (properties != null) {
+      switch (properties.getString("name")) {
+        case "bodytype":
+          setBodyType(
+              properties.get("value").asString().equals("static") ? BodyDef.BodyType.StaticBody
+                  : BodyDef.BodyType.DynamicBody);
+          break;
+        case "density":
+          setDensity(properties.getFloat("value"));
+          break;
+        case "friction":
+          setFriction(properties.getFloat("value"));
+          break;
+        case "restitution":
+          setRestitution(properties.getFloat("value"));
+          break;
+        case "force":
+          setForce(properties.getFloat("value"));
+          break;
+        case "damping":
+          setDamping(properties.getFloat("value"));
+          break;
+        case "maxspeed":
+          setMaxSpeed(properties.getFloat("value"));
+          break;
+        case "debugcolor":
+          try {
+            String cname = properties.getString("value").toUpperCase();
+            Field field = Class.forName("com.badlogic.gdx.graphics.Color").getField(cname);
+            debugColor = new Color((Color) field.get(null));
+          } catch (Exception e) {
+            debugColor = null;
+          }
+          setDebugColor(debugColor);
+          break;
+        case "debugopacity":
+          debugOpacity = properties.getInt("value");
+          setDebugColor(getDebugColor().mul(debugOpacity / 255.0f));
+          break;
+        case "texture":
+          String key = properties.getString("value");
+          TextureRegion texture = new TextureRegion(directory.getEntry(key, Texture.class));
+          setTexture(texture);
+          break;
+        case "sensorcolor":
+          try {
+            String cname = properties.getString("value").toUpperCase();
+            Field field = Class.forName("com.badlogic.gdx.graphics.Color").getField(cname);
+            sensorColor = new Color((Color) field.get(null));
+          } catch (Exception e) {
+            sensorColor = null; // Not defined
+          }
+          break;
+        case "sensoropacity":
+          int opacity = properties.get("value").asInt();
+          if (sensorColor != null) {
+            sensorColor.mul(opacity / 255.0f);
+          }
+          break;
+        default:
+          break;
+      }
+
+      if (debugOpacity != -1 && debugColor != null) {
+        debugColor.mul(debugOpacity / 255f);
+        setDebugColor(debugColor);
+      }
+
+      properties = properties.next();
+    }
+
+  }
+
+
+  /**
+   * Applies the force to the body of this player
+   * <p>
+   * This method should be called after the force attribute is set.
+   */
+  public void applyForce() {
```

Comment:
why did you copy this code from the player model? I am thinking that the NPC model we just want to move at a constant velocity, so we shouldn't have to apply force to it..

Reply:
I think applyforce works better because if the player pushes the NPC off the cliff or anything like that, or if in the future the NPC needs to navigate through some other terrains, this implementation makes more sense because both player & NPC are using the same logic to move around. The movement works nice here with the existing applyForce function so let me know what you think. I tried to change it to set speed but sometimes (if the NPC is not on a complete plain) the physics is not so natural.

Reply:
Or we can just change the entire thing to not a physical body as you commented